### ISSUE TITLE
[Issue] Warning: cherrypy.TimeoutError() TimeoutError

### ISSUE THREAD

**Issue by YipYup** [sabnzbd/sabnzbd#497]
- sabnzbd+ v1.0.0 RC1
- Ubuntu 14.04 LTS 64-bits

I am seeing this in my logs every few days. Not sure if I can back this up with data, but I can usually rectify the symptoms by restarting SAB from CLI. That will usually get me running for a few more days before the symptoms show up again.

`Traceback (most recent call last): File "/usr/share/sabnzbdplus/cherrypy/_cpwsgi.py", line 180, in trap return func(*args, **kwargs) File "/usr/share/sabnzbdplus/cherrypy/_cpwsgi.py", line 99, in __call__ return self.nextapp(environ, start_response) File "/usr/share/sabnzbdplus/cherrypy/_cpwsgi.py", line 410, in tail return self.response_class(environ, start_response, self.cpapp) File "/usr/share/sabnzbdplus/cherrypy/_cpwsgi.py", line 234, in __init__ self.run() File "/usr/share/sabnzbdplus/cherrypy/_cpwsgi.py", line 349, in run request.run(meth, path, qs, rproto, headers, rfile) File "/usr/share/sabnzbdplus/cherrypy/_cprequest.py", line 617, in run raise cherrypy.TimeoutError() TimeoutError`

**Comment by shypike** [comment]
We've had a few of these reports.
So far not enough info to point to a solution.

**Comment by Safihre** [comment]
Was adding 400 NZB's using the Watched folder (for testing how the queue/history-API handles large queues) and then it happend to me.
But the interesting thing is that it happend without me having SAB open anywhere..
```
[25/Mar/2016:12:36:03]  Traceback (most recent call last):
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 180, in trap
    return func(*args, **kwargs)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 99, in __call__
    return self.nextapp(environ, start_response)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 410, in tail
    return self.response_class(environ, start_response, self.cpapp)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 234, in __init__
    self.run()
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 349, in run
    request.run(meth, path, qs, rproto, headers, rfile)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cprequest.py", line 617, in run
    raise cherrypy.TimeoutError()
TimeoutError
```

**Comment by shypike** [comment]
Likely CherryPy has some internal threads which are time-critical. Absorbing 400 NZB files is a very CPU-bound activity which may very well suffocate the other threads.
You'll find that the web UI is inaccessible during those times.
It's a trade-off between waiting between NZB files and allowing the UI to continue or to do it as fast as possible.

**Comment by Safihre** [comment]
I ran the test again but now with #515 branch and logging, and it turns out this error comes at the **end** of adding all these NZB's. 
This is literally the end of my log and besides this showing in the interface, everything is working just fine:
```
2016-03-25 14:14:05,760::INFO::[nzbqueue:251] Saving queue
2016-03-25 14:14:05,785::INFO::[__init__:941] Saving data for queue10.sab in C:\Users\userv\AppData\Local\sabnzbd\admin\queue10.sab
2016-03-25 14:14:06,329::ERROR::[_cplogging:216] [25/Mar/2016:14:14:06]  Traceback (most recent call last):
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 180, in trap
    return func(*args, **kwargs)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 99, in __call__
    return self.nextapp(environ, start_response)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 411, in tail
    return self.response_class(environ, start_response, self.cpapp)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 234, in __init__
    self.run()
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cpwsgi.py", line 350, in run
    request.run(meth, path, qs, rproto, headers, rfile)
  File "C:\Users\userv\Documents\GitHub\sabnzbd\cherrypy\_cprequest.py", line 617, in run
    raise cherrypy.TimeoutError()
TimeoutError
```

Maybe we should just add a catch for this since CherryPy 5 doesn't solve it but it also doesn't inpact performance?

**Comment by thezoggy** [comment]
or maybe increase timeout/deal with large/long queues..

**Comment by thezoggy** [comment]
just helped someone on irc that had this issue after upgrading to 1.0...  their traceback:
https://bpaste.net/show/1526b743f92c

sab running on centos 7, the issue was them trying to save across their network to a raid array with ntfs.. so it had errors saving thus got stuck in an endless loop/crashing and restarting. once i had them change their temp folder to something local.. everything worked smoothly.

**Comment by shypike** [comment]
@thezoggy Any idea which time-out and where?

**Comment by thezoggy** [comment]
it was something on startup where it was trying to save part of a nzb file and it couldnt.. sadly i dont have the full log but still just wanted to mention what we ran into and how we fixed it.

**Comment by YipYup** [comment]
All, I had a thought. I have customized my (CPU) Nice and Disk IO Nice settings. I wonder if others have as well who have run into this error. I have since tweaked my CPU and Disk nice settings from what they were at the time I encountered the error, and I have not seen the problem since then. I don't believe the nice configuration tweaks directly cause the issue, but exacerbate the process blocking if they are configured improperly (as it is possible I had done so initially). I haven't seen the error again since initially opening the issue, but also the conditions under which it appeared haven't been replicated very often -- It is not often I have to restart this service or guest OS.

Here is what I'm running with now:

- Nice:  -n19
- IO Nice:  -c3 -n7
- Extra PAR2 Parameters:  -v -t+
